convert2rhel/toolopts.py

```diff
@@ -515,70 +539,122 @@ def _process_cli_options(self):
-                if header in headers:
-                    for option in config_file.options(header):
-                        if option.lower() in supported_opts:
-                            # Solving priority
-                            if supported_opts[option.lower()] is None:
-                                supported_opts[option] = config_file.get(header, option)
-                                loggerinst.debug("Found %s in %s" % (option, path))
-                        else:
-                            loggerinst.warning("Unsupported option %s in %s" % (option, path))
-                elif header not in headers and header != "DEFAULT":
-                    loggerinst.warning("Unsupported header %s in %s." % (header, path))
-
-    return supported_opts
+    found_opts = {}
+
+    for path in reversed(paths):
+        loggerinst.debug("Checking configuration file at %s" % path)
+        # Check for correct permissions on file
+        if not oct(os.stat(path).st_mode)[-4:].endswith("00"):
+            loggerinst.critical("The %s file must only be accessible by the owner (0600)" % path)
+
+        config_file.read(path)
+
+        # Mapping of all supported options we can have in the config file
+        for supported_header, supported_opts in CONFIG_FILE_MAPPING_OPTIONS.items():
+            loggerinst.debug("Checking for header '%s'" % supported_header)
+            if supported_header not in config_file.sections():
+                loggerinst.warning("Couldn't find header '%s' in the configuration file %s." % (supported_header, path))
+                continue
+
+            options = _get_options_value(config_file, supported_header, supported_opts)
+            found_opts.update(options)
+
+    return found_opts
+
+
+def _get_options_value(config_file, header, supported_opts):
+    """Helper function to iterate through the options in a config file.
+
+    :param config_file: An instance of `py:ConfigParser` after reading the file
+        to iterate through the options.
+    :type config_file: configparser.ConfigParser
+    :param header: The header name to get options from.
+    :type header: str
+    :param supported_opts: List of supported options that can be parsed from
+        the config file.
+    :type supported_opts: list[str]
+    :return: Dict of keys and values loaded from the config file under the provided header
+    :rtype: dict[str, str]
+    """
+    options = {}
+    conf_options = config_file.options(header)
+
+    if len(conf_options) == 0:
+        loggerinst.debug("No options found for %s. It seems to be empty or commented." % header)
+        return options
+
+    for option in conf_options:
+        if option.lower() not in supported_opts:
+            loggerinst.warning("Unsupported option '%s' in '%s'" % (option, header))
+            continue
+
+        options[option] = config_file.get(header, option).strip('"')
+        loggerinst.debug("Found %s in %s" % (option, header))
```

Comment:
@Venefilyn, I'm thinking about the way we save the value associated with the option here... I was about to use a `strip('"')` to remove any trailing `"` we found in the option, but I'm afraid that this could cause problems when dealing with subman secrets. 

Take a look at the following example: 

```
>>> a = '"aushs'
>>> a.strip('"')
'aushs'
>>> a = '"aushs"'
>>> a.strip('"')
'aushs'
>>> a = '"aus"hs"'
>>> a.strip('"')
'aus"hs'
>>> a = '"aus"h"s"'
>>> a.strip('"')
'aus"h"s'
```

When we have a `"` in the middle of the string, we don't accidently remove it, so if the password or any other important secret has a `"` in the middle of it, we would parse normally. Whoever, if the secret starts with only a `"` (as the first example there) we would remove it and make it incorrect. 

Currently, when we parse the value we would store them with the `"` in our dictionary, so the following config:

```config
[subscription_manager]
password = "test12345"
```

would become something like: 

```python
{"subscription_manager": {"password": '"test12345"'}}
```

For that reason, I think it is better to keep the current parser the way it is... What do you think? In case that they have those type of special characters, we would not mess with it, making it more realible maybe? At this point, if something fails, it would be easier to tell the customer that they need to remove the trailing `"` from their string rather than trying to be smart with this, right?

Reply:
@hosekadam what is your input on this as well?

Reply:
I as the customer would be happy if I don't have to care about the formatting much - so we should accept both for having the best user experience.

But I as developer would be also fine with the approach having it as it is and do not make any changes to the values. Just use what is in the config file. And also not adding the complexity to the code.

When I put those feelings together, I think the compromise would be removing the `"` symbol only if it's present in the start and in the end of the string. In that case, it would be a big coincidence if the `"` would be the real part of the string. We can use some regex to check if the is in the start and end and `if`, so I believe it won't increase the complexity of the code.

Anyway, let's wait for another inputs for this.

Reply:
Yeah, after thinking more about that, I really think that we should go with the way it was. It's a bit scary coming up with a sophisticated parser without need, just to guarantee that we are removing correctly `"`. I think that I prefer to leave as-is and parse the user data from that file as they added it. Less room for errors.

Reply:
Maybe you could use `ast.literal_eval()`?

Reply:
Why do we store it as `password = "test12345"` in our config file to begin with? Can't we just get rid of the quotes there. It doesn't make sense to parse anything out, especially not passwords. You could set your own password anywhere that would both start and end with double-quote (")